netlink-packet-route/src/rtnl/route/nlas/next_hops.rs

```diff
@@ -67,57 +75,59 @@ pub struct NextHop {
     pub interface_id: u32,
     /// Gateway address (it is actually encoded as an `RTA_GATEWAY` nla)
     pub gateway: Option<IpAddr>,
```

Comment:
What do you think of removing `gateway`? It feels weird that to be the gateway both in `self.nlas` and `self.gateway`, and I don't think there are other places where we do this.

Reply:
Yes, that seems like a good idea. If we're making a non-backwards-compatible API change, then might as well do it properly.